include/log.h

```diff
@@ -67,6 +67,8 @@ struct ofono_debug_desc {
 					__FILE__, __FUNCTION__ , ## arg); \
 } while (0)
 
+#define PRINTABLE_STR(s) ((s) ? (s) : "(null)")
```

Comment:
Doing printf("%s", a) with a==NULL is actually undefined behaviour following the C standard. For instance, printf("%s\n", NULL) segfaults in glibc [1]. Although glibc is clever enough to print "(null)" instead of segfaulting in *most* cases, still, trying to print NULL is not good C. So it is better to assume printf cannot handle NULL pointers (and behaviour is not consistent across the printf family, see comment 6 in [2]).  So I prefer to use this macro.

[1] https://sourceware.org/bugzilla/show_bug.cgi?id=10818
[2] https://gcc.gnu.org/bugzilla/show_bug.cgi?id=15685